### PR TITLE
fix: updated public access endpoint

### PR DESCRIPTION
**Pull request by iamar7**
### Description

The resource restapi_object.account_public_access in main.tf was using public endpoint. Updated it to support private endpoint.

https://github.com/terraform-ibm-modules/terraform-ibm-iam-account-settings/issues/86

#### Changes that affect the core Terraform module or submodules
- [ ] Bug fix
- [ ] New feature
- [ ] Dependency update

#### Changes that don't affect the core Terraform module or submodules
- [ ] Examples or tests (addition or updates of examples or tests)
- [ ] Documentation update
- [ ] CI-related update (pipeline, etc.)
- [ ] Other

#### Release required?
Identify the type of release. For information about the changes in a semantic versioning release, see [Release versioning](https://terraform-ibm-modules.github.io/documentation/#/versioning).

- [ ] No release
- [ ] Patch release (`x.x.X`)
- [ ] Minor release (`x.X.x`)
- [ ] Major release (`X.x.x`)

##### Release notes content

The path of the resource restapi_object.account_public_access is updated to support private endpoint.

### Run the pipeline

If the CI pipeline doesn't run when you create the PR, the PR requires a user with GitHub collaborators access to run the pipeline.

Run the CI pipeline when the PR is ready for review and you expect tests to pass. Add a comment to the PR with the following text:

```
/run pipeline
```

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### Merge actions for mergers

- Use a relevant [conventional commit](https://www.conventionalcommits.org/) message that is based on the PR contents and any release notes provided by the PR author. The commit message determines whether a new version of the module is needed, and if so, which semver increment to use (major, minor, or patch).
- Merge by using "Squash and merge".

### CODE REVIEW

**Comment by ocofaigh** [comment]
Spotted a bug that is causing pipeline to fail. Fixed in https://github.com/terraform-ibm-modules/terraform-ibm-iam-account-settings/pull/107

File: main.tf [hunk 1]
```diff
@@ -15,10 +15,11 @@ data "ibm_cloud_shell_account_settings" "cloud_shell_account_settings" {
...
 resource "ibm_iam_account_settings" "iam_account_settings" {
-  if_match                                   = "*"
-  allowed_ip_addresses                       = local.iam_allowed_ip_addresses
-  max_sessions_per_identity                  = var.max_sessions_per_identity
-  mfa                                        = var.mfa
+  if_match                  = "*"
+  allowed_ip_addresses      = local.iam_allowed_ip_addresses
+  max_sessions_per_identity = var.max_sessions_per_identity
+  mfa                       = var.mfa
+  # private_endpoint = var.private_endpoint
```
Comment (ocofaigh) [inline main.tf:22]: remove this comment

File: main.tf [hunk 2]
```diff
@@ -37,7 +38,7 @@ resource "ibm_cloud_shell_account_settings" "cloud_shell_account_settings" {
 # Configure account public access
 # (Using restapi provider for this until official IBM provider support is added -> https://github.com/IBM-Cloud/terraform-provider-ibm/issues/3285)
 resource "restapi_object" "account_public_access" {
-  path           = "//iam.cloud.ibm.com/v2/groups/settings?account_id={id}"
+  path           = var.private_endpoint == false ? "//private.iam.cloud.ibm.com/v2/groups/settings?account_id={id}" : "//iam.cloud.ibm.com/v2/groups/settings?account_id={id}"
```
Comment (ocofaigh) [inline main.tf:41]: Isn't this logic backwards?

File: variables.tf [hunk 3]
```diff
@@ -26,6 +26,12 @@ variable "public_access_enabled" {
   default     = false
 }
 
+variable "private_endpoint" {
+  type        = bool
+  description = "To enable/disable the private endpoint in the account public access."
```
Comment (ocofaigh) [inline variables.tf:31]: Set to true to use the private IAM endpoint which is used to configure the public access setting

**Review by ocofaigh — CHANGES_REQUESTED** [review]
see comments

File: main.tf (hunk 1 above)
Comment (iamar7) [reply main.tf:22]: Done

File: variables.tf (hunk 3 above)
Comment (iamar7) [reply variables.tf:31]: Done.

File: main.tf (hunk 2 above)
Comment (iamar7) [reply main.tf:41]: Changed it to 
`path           = var.private_endpoint ? "//private.iam.cloud.ibm.com/v2/groups/settings?account_id={id}" : "//iam.cloud.ibm.com/v2/groups/settings?account_id={id}"`

File: variables.tf (hunk 3 above)
Comment (ocofaigh) [reply variables.tf:31]: This does not match what I said in my comment. I would also default it to back to `false`

File: main.tf [hunk 4]
```diff
@@ -37,7 +37,7 @@ resource "ibm_cloud_shell_account_settings" "cloud_shell_account_settings" {
 # Configure account public access
 # (Using restapi provider for this until official IBM provider support is added -> https://github.com/IBM-Cloud/terraform-provider-ibm/issues/3285)
 resource "restapi_object" "account_public_access" {
-  path           = "//iam.cloud.ibm.com/v2/groups/settings?account_id={id}"
+  path           = var.private_endpoint ? "//private.iam.cloud.ibm.com/v2/groups/settings?account_id={id}" : "//iam.cloud.ibm.com/v2/groups/settings?account_id={id}"
```
Comment (ocofaigh) [inline main.tf:40]: can we move the code duplication here? Put `iam.cloud.ibm.com/v2/groups/settings?account_id={id}` into a local variable

Comment (iamar7) [reply main.tf:40]: Done

File: variables.tf (hunk 3 above)
Comment (iamar7) [reply variables.tf]: Sorry my bad. Updated the description and changed the default value to `false`.

**Review by ocofaigh — APPROVED** [review]

**Pull request merged**